domain/miningmanager/mempool/model/ordered_transactions_by_fee_rate.go

```diff
@@ -0,0 +1,80 @@
+package model
+
+import (
+	"sort"
+
+	"github.com/pkg/errors"
+)
+
+// TransactionsOrderedByFeeRate represents a set of MempoolTransactions ordered by their fee / mass rate
+type TransactionsOrderedByFeeRate struct {
+	slice []*MempoolTransaction
+}
+
+// GetByIndex returns the transaction in the given index
+func (tobf *TransactionsOrderedByFeeRate) GetByIndex(index int) *MempoolTransaction {
```

Comment:
This seems unnecessary if the only thing you ever do is getting the first element.
Please consider replacing this function with something like First().

Reply:
I wanted the type to be more general then how it's currently used, but anyway, I'm now using this for non-first index as well.